From 7163234c90069fe287ebb0435e7edab8c5978ff0 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Tue, 22 Sep 2020 20:57:00 +0530
Subject: [PATCH 1606/1851] dt-bindings: edac: Add bindings for Xilinx Versal
 EDAC for DDRMC

commit 6f25fc9d2cef8505fe64ff12a7e7b347acfbe8a7 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds device tree bindings for Xilinx Versal EDAC for DDR
controller.

Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/edac/xlnx,versal-ddrmc-edac.yaml | 60 +++++++++++++++++++
 1 file changed, 60 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/edac/xlnx,versal-ddrmc-edac.yaml

diff --git a/Documentation/devicetree/bindings/edac/xlnx,versal-ddrmc-edac.yaml b/Documentation/devicetree/bindings/edac/xlnx,versal-ddrmc-edac.yaml
new file mode 100644
index 000000000000..b14fcb4f6201
--- /dev/null
+++ b/Documentation/devicetree/bindings/edac/xlnx,versal-ddrmc-edac.yaml
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: GPL-2.0
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/edac/xlnx,versal-ddrmc-edac.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Xilinx Versal DDRMC EDAC driver
+
+maintainers:
+  - Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
+
+description: |
+  Versal DDR controller has an optional ECC support which correct
+  single bit ECC errors and detect double bit ECC errors.
+
+properties:
+  compatible:
+    items:
+      - enum:
+          - xlnx,versal-ddrmc-edac
+
+  reg:
+    maxItems: 2
+    items:
+      - description: DDR Memory Controller registers
+      - description: NOC registers corresponding to DDR Memory Controller
+
+  reg-names:
+    items:
+      - const: ddrmc_base
+      - const: ddrmc_noc_base
+
+  interrupts:
+    maxItems: 1
+
+  xlnx,mc-id:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description: Instance number of the DDR Memory Controller
+
+required:
+  - compatible
+  - reg
+  - reg-names
+  - interrupts
+  - xlnx,mc-id
+
+examples:
+  - |
+    bus {
+      #address-cells = <2>;
+      #size-cells = <2>;
+      memory-controller@f6150000 {
+        compatible = "xlnx,versal-ddrmc-edac";
+        reg = <0x0 0xf6150000 0x0 0x2000>, <0x0 0xf6070000 0x0 0x20000>;
+        reg-names = "ddrmc_base", "ddrmc_noc_base";
+        interrupt-parent = <&gic>;
+        interrupts = <0 147 4>;
+        xlnx,mc-id = <0>;
+      };
+    };
-- 
2.31.1

